lib/netext/dialer.go

```diff
@@ -66,31 +65,42 @@ func (b BlackListedIPError) Error() string {
 	return fmt.Sprintf("IP (%s) is in a blacklisted range (%s)", b.ip, b.net)
 }
 
+func resolveHost(host string, hosts map[string]net.IP, resolver *dnscache.Resolver) (net.IP, error) {
```

Comment:
A minor nitpick, but if this was a method of the `Dialer`, you can skip the first 2 arguments while still having it very testable. And mocking it can easily be done by just defining a simple new `interface` with a `FetchOne()` method yourself, for now (there will probably be further methods once we fix the DNS issues) and using that in the `Dialer`.

Reply:
Thanks, I did that in d6c6ecc0, but only the `hosts` argument can be removed, we need the other 2 :)